qupath-gui-fx/src/main/java/qupath/lib/gui/commands/Commands.java

```diff
@@ -1767,6 +1723,11 @@ public static void convertDetectionsToPoints(ImageData<?> imageData, boolean pre
 		
 		boolean	deleteDetections = button == ButtonType.YES;
 		PathObjectTools.convertToPoints(hierarchy, pathObjects, preferNucleus, deleteDetections);
+		imageData.getHistoryWorkflow().addStep(
+				new DefaultScriptableWorkflowStep("Convert detections to points",
+						String.format("PathObjectTools.convertToPoints(getCurrentHierarchy(), getDetectionObjects(), %b, %b)", preferNucleus, deleteDetections)
```

Comment:
Related to the point below, is this important enough that we could justify adding methods to `QP`?

The methods would warrant some thought, since `QP` is already very bloated. The use of two booleans isn't great as no one will remember what they mean.
Elsewhere, we've used [varargs options](https://github.com/qupath/qupath/blob/60898ed353b2eda253d0f0800232204bde11ecc3/qupath-core-processing/src/main/java/qupath/lib/scripting/QP.java#L2960) to make the custom options more explicit in scripts.

Reply:
I think `deleteObjects` can be `true` because the objects aren't really deleted (just removed from the hierarchy); people can easily keep a handle on the objects if they need to, and it is imo what people will expect the command to do anyways.

I'd also be fine setting `preferNucleus` to be true; it's currently set to true in the GUI anyways, and it's not much more code to directly call the PathObjectTools method if you really want to muck about

Reply:
Mmmm, not that easy to keep the detections - even retaining a reference isn't enough since they'll have parents & reinserting them into the object hierarchy won't necessarily put them back where they started.

Dropping `preferNucleus` is appealing. We'd need to get it out of the method signature to ensure we can't log a version of the command that is different from what was actually called.

Reply:
Suggest we continue discussion in #1751